common/src/autogluon/common/utils/try_import.py

```diff
@@ -30,7 +31,7 @@ def try_import_mxboard():
 
 
 def try_import_ray() -> ModuleType:
-    RAY_MAX_VERSION = "2.7.0"
+    RAY_MAX_VERSION = "2.7.0"  # sync with core/setup.py
```

Comment:
have some updates here as well on my branch, some critocal fixes for ray serve specific to py 3.11 were shipped with 2.7.0

Reply:
ray 2.7 has breaking changes that will break our existing code. I think it's better to have a separate PR for the ray upgrade. I feel it would be a bit messy to have all changes being squashed into this PR